tfhe/src/integer/ciphertext/compact_list.rs

```diff
@@ -1,42 +1,380 @@
+        num_blocks: usize,
+    ) -> &mut Self
+    where
+        T: Compactable + Numeric,
+    {
+        for value in values {
+            self.push_with_num_blocks(value, num_blocks);
+        }
+        self
+    }
+
+    pub fn build(&self) -> CompactCiphertextList {
+        let ct_list = self.pk.key.encrypt_slice(self.messages.as_slice());
+        CompactCiphertextList {
+            ct_list,
+            info: self.info.clone(),
+        }
+    }
+
+    pub fn build_packed(&self) -> CompactCiphertextList {
+        // Here self.messages are decomposed blocks in range [0..message_modulus[
+        let msg_mod = self.pk.key.message_modulus().0 as u64;
+        let packed_messaged_iter = self
+            .messages
+            .chunks(2)
+            .map(|two_values| (two_values[1] * msg_mod) + two_values[0]);
+        let ct_list = self
+            .pk
+            .key
+            .encrypt_iter_with_modulus(packed_messaged_iter, msg_mod * msg_mod);
+
+        CompactCiphertextList {
+            ct_list,
+            info: self.info.clone(),
+        }
+    }
+
+    /// For zk proofs, message are always packed
+    #[cfg(feature = "zk-pok-experimental")]
+    pub fn build_with_proof(
+        &self,
+        public_params: &CompactPkePublicParams,
+        load: ZkComputeLoad,
+    ) -> crate::Result<ProvenCompactCiphertextList> {
+        let ct_list = self.pk.key.encrypt_and_prove_slice(
+            self.messages.as_slice(),
+            public_params,
+            load,
+            self.pk.key.parameters.message_modulus().0 as u64,
+        )?;
+        Ok(ProvenCompactCiphertextList {
+            ct_list,
+            info: self.info.clone(),
+        })
+    }
+
+    #[cfg(feature = "zk-pok-experimental")]
+    pub fn build_with_proof_packed(
+        &self,
+        public_params: &CompactPkePublicParams,
+        load: ZkComputeLoad,
+    ) -> crate::Result<ProvenCompactCiphertextList> {
+        let msg_mod = self.pk.key.message_modulus().0 as u64;
+        let packed_messages = self
+            .messages
+            .chunks(2)
+            .map(|two_values| (two_values[1] * msg_mod) + two_values[0])
+            .collect::<Vec<_>>();
+        let ct_list = self.pk.key.encrypt_and_prove_slice(
+            packed_messages.as_slice(),
+            public_params,
+            load,
+            msg_mod * msg_mod,
+        )?;
+        Ok(ProvenCompactCiphertextList {
+            ct_list,
+            info: self.info.clone(),
+        })
+    }
+}
+
+pub trait Expandable: Sized {
+    fn from_expanded_blocks(blocks: &[Ciphertext], kind: DataKind) -> crate::Result<Self>;
+}
+
+impl<T> Expandable for T
+where
+    T: IntegerRadixCiphertext,
+{
+    fn from_expanded_blocks(blocks: &[Ciphertext], kind: DataKind) -> crate::Result<Self> {
+        match (kind, T::IS_SIGNED) {
+            (DataKind::Unsigned(_), false) | (DataKind::Signed(_), true) => {
+                Ok(T::from_blocks(blocks.to_vec()))
+            }
```

Comment:
No because the integer API is not typed with regards to the size of the ciphertexts as the HLAPI is.

So the user has to extract the correct kind, and if the circuit really requires specific sizes of radixes, the user
can check the number of blocks in the resulting ciphertext